test/lib/Instrumentation/CMakeLists.txt

```diff
@@ -24,4 +24,8 @@ foreach( plugin ${GPU_INSTRUMENTATION_PASSES} )
      # which causes the llvmGetPassPluginInfo symbol to be hidden and
      # an "entry point not found" error. Reset it just for this target
      target_compile_options(${plugin} PRIVATE -fvisibility=default)
+
+     set_target_properties(${plugin} PROPERTIES
+     	     LIBRARY_OUTPUT_DIRECTORY "${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/../instrumentation")
```

Comment:
We don't have issues if installing from pip directly; but this will try to create `/instrumentation` directory if `CMAKE_LIBRARY_OUTPUT_DIRECTORY` is not set? Maybe wrap `set_target_properties` with `if (CMAKE_LIBRARY_OUTPUT_DIRECTORY)`?

Reply:
I think CMAKE_LIBRARY_OUTPUT_DIRECTORY that will always be set as an internal CMake variable. I suppose another option is to use CMAKE_BINARY_DIR as anchor location which I know for a fact is always set but it makes the logic more convoluted.

Reply:
It's explicitly set when [building from python](https://github.com/triton-lang/triton/blob/main/python/setup.py#L417). I typically do not build from python for just compiler changes--I built directly from the root `CMakeLists.txt`. Right now it's erroring out for me because the default `CMAKE_LIBRARY_OUTPUT_DIRECTORY` value is empty.

Reply:
Got it. That makes sense.